### PR TITLE
Issue/migrate to microfrontendplatform v1.0.0 beta.20

### PR DESCRIPTION
**Pull request by danielwiehl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [guidelines](https://github.com/SchweizerischeBundesbahnen/scion-workbench/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added
- [x] Docs have been added or updated

## PR Type
What kind of change does this PR introduce?

- [ ] Fix
- [x] Feature
- [ ] Documentation
- [ ] Refactoring (changes that neither fixes a bug nor adds a feature)
- [ ] Performance (changes that improve performance)
- [ ] Test (adding missing tests, refactoring tests; no production code change)
- [ ] Chore (other changes like formatting, updating the license, updating dependencies, removal of deprecations, etc)
- [ ] CI (changes to our CI configuration files and scripts)
- [ ] Revert (revert of a previous commit)
- [ ] Release (publish a new release)
- [ ] Other... Please describe:

## Issue

Issue Number:
- SchweizerischeBundesbahnen/scion-microfrontend-platform#96

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

**BREAKING CHANGE**: Supporting `@scion/microfrontend-platform v1.0.0-beta.20` introduced a breaking change in the configuration of the host application and the host/client communication protocol.

SCION Microfrontend Platform consolidated the API for configuring the platform, eliminating the different ways to configure the platform. Consequently, SCION Workbench could also simplify its API for enabling microfrontend support.

Related issues of the SCION Microfrontend Platform:
- SchweizerischeBundesbahnen/scion-microfrontend-platform#39
- SchweizerischeBundesbahnen/scion-microfrontend-platform#96

#### Host App Migration:
- property `WorkbenchModuleConfig.microfrontends` has been renamed to `WorkbenchModuleConfig.microfrontendPlatform` and its type changed from `WorkbenchMicrofrontendConfig` to `MicrofrontendPlatformConfig` (provided by @scion/microfrontend-platform);
- `MicrofrontendPlatformConfigLoader` has been changed to return an instance of `MicrofrontendPlatformConfig` instead of the `PlatformConfig`;
- DI token `POST_MICROFRONTEND_PLATFORM_CONNECT` has been renamed to `MICROFRONTEND_PLATFORM_POST_STARTUP` in order to be consistent with other workbench DI tokens;
- provide the host's manifest, if any, via `MicrofrontendPlatformConfig.host.manifest` instead of `WorkbenchMicrofrontendConfig.platformHost.manifest`; either as URL or object literal
- register applications in `MicrofrontendPlatformConfig.applications` instead of `WorkbenchMicrofrontendConfig.platform.apps`;
- specify the symbolic name of the host in `MicrofrontendPlatformConfig.host.symbolicName` instead of `WorkbenchMicrofrontendConfig.platformHost.symbolicName`;
- configure properties in `MicrofrontendPlatformConfig.properties` instead of `WorkbenchMicrofrontendConfig.platform.properties`;
- specify global `manifestLoadTimeout` in `MicrofrontendPlatformConfig.manifestLoadTimeout` instead of `WorkbenchMicrofrontendConfig.platform.manifestLoadTimeout`;
- specify global `activatorLoadTimeout` in `MicrofrontendPlatformConfig.activatorLoadTimeout` instead of `WorkbenchMicrofrontendConfig.platform.activatorLoadTimeout`;
- the bean `MicroApplicationConfig` has been removed; you can now obtain the application's symbolic name as following: `Beans.get<string>(APP_IDENTITY)`;
- the interface `ApplicationManifest` has been renamed to `Manifest`;

For further instructions on how to migrate the host, refer to https://github.com/SchweizerischeBundesbahnen/scion-microfrontend-platform/blob/master/docs/site/changelog/changelog.md#host-app-migration

#### Client App Migration
- the micro application must now pass its identity (symbolic name) directly as the first argument, rather than via the options object;
- the options object passed to `WorkbenchClient.connect` has been renamed from ` MicroApplicationConfig` to `ConnectOptions` and messaging options are now top-level options;
- the bean `MicroApplicationConfig` has been removed; you can now obtain the application's symbolic name as following: `Beans.get<string>(APP_IDENTITY)`;

For further instructions on how to migrate the client, refer to https://github.com/SchweizerischeBundesbahnen/scion-microfrontend-platform/blob/master/docs/site/changelog/changelog.md#client-app-migration

### CODE REVIEW

**Review by mofogasy — CHANGES_REQUESTED** [review]
Some minor findings and questions.

File: projects/scion/workbench/package.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "@angular/animations": "^12.0.0-rc.0 || ^12.0.0",
     "@angular/forms": "^12.0.0-rc.0 || ^12.0.0",
     "@angular/router": "^12.0.0-rc.0 || ^12.0.0",
-    "@scion/toolkit": ">=12.0.0-beta.1",
-    "@scion/microfrontend-platform": ">=1.0.0-beta.16",
-    "@scion/workbench-client": ">=1.0.0-beta.7",
+    "@scion/toolkit": ">=12.0.0-beta.3",
```
[the remark this answers is not in the document]
Comment (danielwiehl) [reply projects/scion/workbench/package.json:27]: Not necessarily. The workbench can require a higher version than `@scion/workbench-client`

File: projects/scion/workbench/src/lib/startup/workbench-initializer.ts [hunk 2]
```diff
@@ -62,17 +71,17 @@ export const WORKBENCH_POST_STARTUP = new InjectionToken<WorkbenchInitializer |
  * executed.
  *
```
[the remark this answers is not in the document]
Comment (danielwiehl) [reply projects/scion/workbench/src/lib/startup/workbench-initializer.ts:72]: done

**Review by mofogasy — APPROVED** [review]
LGTM

**Pull request merged**